localstack/testing/pytest/fixtures.py

```diff
@@ -612,15 +612,17 @@ def _create_job(audio_file: str, params: Optional[dict[str, Any]] = None) -> str
 
 
 @pytest.fixture
-def kinesis_create_stream(aws_client):
+def kinesis_create_stream(aws_client, wait_for_stream_ready):
     stream_names = []
 
     def _create_stream(**kwargs):
         if "StreamName" not in kwargs:
             kwargs["StreamName"] = f"test-stream-{short_uid()}"
+        stream_name = kwargs["StreamName"]
         aws_client.kinesis.create_stream(**kwargs)
-        stream_names.append(kwargs["StreamName"])
-        return kwargs["StreamName"]
+        stream_names.append(stream_name)
+        wait_for_stream_ready(stream_name)
+        return stream_name
```

Comment:
Makes sense.
The fixture is not used in -ext but a similar (now same) one is implemented in -ext. 
My suggestion would be to refactor the fixture like so, this would allow to remove the identical fixture from -ext to remove code duplication.

Reply:
The one in -ext also does not wait for the stream ready state before returning, but before deleting. I agree with consolidating them, but maybe in a separate PR.

Reply:
resolved in: 8c9a440e3edde095316e77c790b329b0a5ea8314